adapters/klab.ogc/src/main/java/org/integratedmodelling/klab/stac/STACEncoder.java

```diff
@@ -122,22 +125,25 @@ public void getEncodedData(IResource resource, Map<String, String> urnParameters
                     (int) grid.getXCells(), (int) grid.getYCells());
 
             Integer srid = items.get(0).getEpsg();
-            CoordinateReferenceSystem outputCrs = CrsUtilities.getCrsFromSrid(srid);
+            CrsUtilities.getCrsFromSrid(srid);
             ReferencedEnvelope regionEnvelope = new ReferencedEnvelope(region.toEnvelope(),
-                    DefaultGeographicCRS.WGS84).transform(outputCrs, true);
+                    space.getProjection().getCoordinateReferenceSystem());
             RegionMap regionTransformed = RegionMap.fromEnvelopeAndGrid(regionEnvelope, (int) grid.getXCells(), (int) grid.getYCells());
             String assetId = resource.getParameters().get("asset", String.class);
-            
-            
-            // TODO Inigo check this. I think this needs some discussion. Allow transform 
-            // ensures the process to finish, but I would not bet on the resulting data.
-            boolean allowTransform = true;
-            HMRaster outRaster = HMStacCollection.readRasterBandOnRegion(regionTransformed, assetId, items, true, lpm);
+
+            Set<Integer> ESPGsAtItems = items.stream().map(i -> i.getEpsg()).collect(Collectors.toUnmodifiableSet());
+            if (ESPGsAtItems.size() > 1) {
+                scope.getMonitor().warn("Multiple ESPGs found on the items " + ESPGsAtItems.toString() + ". The transformation process could affect the data.");
+            }
+
+            // Allow transform ensures the process to finish, but I would not bet on the resulting data.
+            final boolean allowTransform = true;
+            HMRaster outRaster = HMStacCollection.readRasterBandOnRegion(regionTransformed, assetId, items, allowTransform, lpm);
```

Comment:
Is it ensured that regionTransformed here is in epsg:4326? To my knowledge it might not be like that if in a different environment than the kexplorer. 
To be sure I would do a transform of the envelope to 4326. What do you think?

Reply:
`regionTransformed` does have a different ESPG, yes. I will check if we can make it 4326 at this step and see the results